pubsub.go

```diff
@@ -483,26 +495,35 @@ func (c *PubSub) initPing() {
 		timer.Stop()
 
 		healthy := true
+		timeout := pingTimeout
 		for {
-			timer.Reset(pingTimeout)
+			timer.Reset(timeout)
 			select {
 			case <-c.ping:
 				healthy = true
+				timeout = pingTimeout
 				if !timer.Stop() {
 					<-timer.C
 				}
 			case <-timer.C:
-				pingErr := c.Ping(ctx)
+				var healthyErr error
+
 				if healthy {
+					healthyErr = c.Ping(ctx)
 					healthy = false
 				} else {
-					if pingErr == nil {
-						pingErr = errPingTimeout
-					}
+					healthyErr = errPingTimeout
+				}
+
+				if healthyErr != nil {
 					c.mu.Lock()
-					c.reconnect(ctx, pingErr)
-					healthy = true
+					c.reconnect(ctx, healthyErr)
 					c.mu.Unlock()
+
+					healthy = true
+					timeout = pingTimeout
+				} else {
+					timeout = c.healthTimeout
```

Comment:
does this make sense -- should we always  use the time that was explicitly set ? 

                                 if healthyErr != nil {
					c.mu.Lock()
					c.reconnect(ctx, healthyErr)
					c.mu.Unlock()
					healthy = true
				} 
				timeout = c.healthTimeout

Reply:
It is the waiting time of the ping command.